src/flags/salesforceId.ts

```diff
@@ -49,18 +49,22 @@ export const salesforceIdFlag = Flags.custom<string, IdFlagConfig>({
 });
 
 const validate = (input: string, config?: IdFlagConfig): string => {
-  const { length, startsWith } = config ?? {};
-  if (length && input.length !== length) {
-    throw messages.createError('errors.InvalidIdLength', [length]);
-  }
-  if (!length && ![15, 18].includes(input.length)) {
-    throw messages.createError('errors.InvalidIdLength', ['15 or 18']);
+  const resolvedConfig = config ?? {};
+
+  resolvedConfig.length = resolvedConfig.length ?? 'both';
+
+  const length = resolvedConfig.length === 'both' ? [15, 18] : [resolvedConfig.length];
+
+  if (!length.includes(input.length)) {
+    throw messages.createError('errors.InvalidIdLength', [
+      length.join(` ${messages.getMessage('errors.InvalidIdLength.or')} `),
+    ]);
```

Comment:
Simplified this a little bit to be easier to follow the logic.

```suggestion
  const { length, startsWith } = config ?? {};

  // If the flag doesn't specify a length or specifies "both", then let it accept both 15 or 18.
  let allowedIdLength = (!length || length === 'both') ? [15, 18] : [length];

  if (!allowedIdLength.includes(input.length)) {
    throw messages.createError('errors.InvalidIdLength', [
      allowedIdLength.join(` ${messages.getMessage('errors.InvalidIdLength.or')} `),
    ]);
```